scripts/build/types.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package build
+
+import (
+	"os"
+)
+
+// Config contains various pieces of configuration created by main and passed
+// around.
+type Config struct {
+	ArchiveFormat string
+	BuildFormats  []string
+	OutputPath    string
+	Packages      []string
+	Run           bool
+	Verbose       bool
+}
+
+// Intermediate files are stored in memory so you do not need sudo to create
+// cpio files. Some fields such as atime and mtime are not included here
+// because they are useless for our purposes.
+type file struct {
```

Comment:
Later is in about six weeks :-) Right now this is designed around a 'put it all in memory' model. If it gets done as a streaming model now, then later redesign won't be needed. Further, we won't have to worry about lots of rework later if we get it right now. 

Or if it gets done as the mmap model, redesign won't be needed. It might be good to write it now so that we don't have to redo a lot in the next month ... The interns this summer will be using this to build root file systems. Note the passthrough model of cpio could make a stream model pretty easy. But mmap is pretty attractive too, the only issue being making the file grow as needed.